### PR TITLE
mgr/dashboard: Improve RgwUser controller

### PR DESCRIPTION
**Pull request by votdev**
Use the new 'user?list' RGW Admin OPS API endpoint.

Signed-off-by: Volker Theile <vtheile@suse.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by LenzGr** [comment]
Please prefix your git commit message with `mgr/dashboard:`

File: src/pybind/mgr/dashboard/controllers/rgw.py [hunk 1]
```diff
@@ -168,7 +168,18 @@ def _append_uid(self, user):
         return user
 
     def list(self):
-        return self.proxy('GET', 'metadata/user')
+        users = []
+        marker = None
+        while True:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/controllers/rgw.py]: Can we make sure, this is not going into an endless loop?

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard/controllers/rgw.py]: Guess my biggest fear is we will end up always use the same marker over and over again.

**Comment by s0nea** [comment]
Unfortunately there is an issue with the QA tests but I don't think it's related to this PR.

Anyway - to make sure we ran all test on the PR we'll need to fix the issue and run the tests again I'm afraid :/

```
2019-01-07T13:13:25.291 INFO:tasks.cephfs_test_runner:======================================================================
2019-01-07T13:13:25.291 INFO:tasks.cephfs_test_runner:FAIL: test_full_health (tasks.mgr.dashboard.test_health.HealthTest)
2019-01-07T13:13:25.292 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-01-07T13:13:25.292 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-01-07T13:13:25.292 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-tdehler-testing-25300/qa/tasks/mgr/dashboard/test_health.py", line 253, in test_full_health
2019-01-07T13:13:25.292 INFO:tasks.cephfs_test_runner:    self.assertSchema(data, schema)
2019-01-07T13:13:25.292 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-tdehler-testing-25300/qa/tasks/mgr/dashboard/helper.py", line 337, in assertSchema
2019-01-07T13:13:25.292 INFO:tasks.cephfs_test_runner:    self.assertEqual(data, str(e))
2019-01-07T13:13:25.293 INFO:tasks.cephfs_test_runner:AssertionError: ..."
```

- https://shaman.ceph.com/builds/ceph/wip-tdehler-testing-25300/926c0cdbb8d8b81cc16f379b480ea66ecca7013c/
- http://pulpito.ceph.com/tdehler-2019-01-07_12:42:53-rados:mgr-wip-tdehler-testing-25300-distro-basic-smithi/

**Comment by s0nea** [comment]
It seems like there's already a fix for it: https://github.com/ceph/ceph/pull/25813
I'm going to run the tests again.

**Comment by s0nea** [comment]
Unfortunately the last QA test run failed:

```
2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:======================================================================
2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:ERROR: test_invalid_user_id (tasks.mgr.dashboard.test_rgw.RgwApiCredentialsTest)
2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-tdehler-testing-25300-25813/qa/tasks/mgr/dashboard/test_rgw.py", line 119, in test_invalid_user_id
2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:    data['message'])
2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:  File "/usr/lib/python2.7/unittest/case.py", line 800, in assertIn
2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:    if member not in container:
2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:TypeError: argument of type 'NoneType' is not iterable
```

- http://pulpito.ceph.com/tdehler-2019-01-08_06:11:53-rados:mgr-wip-tdehler-testing-25300-25813-distro-basic-smithi/
- https://shaman.ceph.com/builds/ceph/wip-tdehler-testing-25300-25813/

**Comment by votdev** [comment]
> Unfortunately the last QA test run failed:
> 
> ```
> 2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:======================================================================
> 2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:ERROR: test_invalid_user_id (tasks.mgr.dashboard.test_rgw.RgwApiCredentialsTest)
> 2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
> 2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
> 2019-01-08T11:35:54.814 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-tdehler-testing-25300-25813/qa/tasks/mgr/dashboard/test_rgw.py", line 119, in test_invalid_user_id
> 2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:    data['message'])
> 2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:  File "/usr/lib/python2.7/unittest/case.py", line 800, in assertIn
> 2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:    if member not in container:
> 2019-01-08T11:35:54.815 INFO:tasks.cephfs_test_runner:TypeError: argument of type 'NoneType' is not iterable
> ```
> 
>     * [http://pulpito.ceph.com/tdehler-2019-01-08_06:11:53-rados:mgr-wip-tdehler-testing-25300-25813-distro-basic-smithi](http://pulpito.ceph.com/tdehler-2019-01-08_06:11:53-rados:mgr-wip-tdehler-testing-25300-25813-distro-basic-smithi/)
> 
>     * [shaman.ceph.com/builds/ceph/wip-tdehler-testing-25300-25813](https://shaman.ceph.com/builds/ceph/wip-tdehler-testing-25300-25813/)

Found the problem. It seems the civetweb configuration has been changed and our parser failed to parse the RGW host.

**Comment by s0nea** [comment]
The last QA test run was successful :)

- https://shaman.ceph.com/builds/ceph/wip-tdehler-testing-25300-25813/5379a14295973b4a384ded82c1f5dc6def575044/
- http://pulpito.ceph.com/tdehler-2019-01-09_09:51:56-rados:mgr-wip-tdehler-testing-25300-25813-distro-basic-smithi/

**Pull request merged**